Теперь в биодоме клоунов появляется палочка поскальзывания вместо посоха

NightDawnFox opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
Добавляет в игру новый предмет - палочку поскальзывания. Имеет 5 зарядов, обладает особенностью - как только истратятся все заряды, способна восстанавливать исключительно один заряд в течение 30 секунд. Больше одного заряда не восстановится. Заменяет посох поскальзывания в биодоме клоунов на палочку. Добавляет уникальные спрайты палочке.
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. -->

## Ссылка на предложение/Причина создания ПР
https://discord.com/channels/617003227182792704/755125334097133628/1120399028883042315
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->

## Демонстрация изменений
https://www.youtube.com/watch?v=cZO5CMRb5YU
<!-- Здесь вы можете показать изменения внешне, к примеру новые спрайты, звуки или изменения карты. Или написать, что именно изменилось для игроков. Этот пункт полностью опционален и его можно удалить. -->

NightDawnFox committed:
That's all!

Dimach commented on code/modules/projectiles/guns/magic/wand.dm:
```diff
@@ -162,3 +162,34 @@
+/obj/item/gun/magic/wand/slipping/afterattack(atom/target, mob/living/user)
+	. = ..()
+	if(!charges)
+		to_chat(usr, "<span class='notice'>[src] is regaining his charge. Be patient!</span>")
```
Добавь какую нибудь проверку на то что палочка в данный момент перезаряжается, потому что иначе я могу потратить все заряды и потом в течении 30 секунд 50 раз наспамить атакой и нагенерить 50 зарядов

NightDawnFox replied:
Так там же во флагах таймера стоит "TIMER_UNIQUE". Если я правильно понял его суть, он как раз таки не даёт спамить зарядами,  ибо таймер срабатывает только один раз

NightDawnFox replied:
https://www.youtube.com/watch?v=K8neZQkBxDw
Единственное раздражающее - повторяющиеся сообщения о перезарядке. Сейчас попробую исправить

Gottfrei replied:
> Так там же во флагах таймера стоит "TIMER_UNIQUE". Если я правильно понял его суть, он как раз таки не даёт спамить зарядами, ибо таймер срабатывает только один раз

нет, "TIMER_UNIQUE", при добавлении его на тот же объект (с теми же аргументами), не позволит создать новый таймер, если старый таймер ещё не закончился, конечно.

тебе наоборот надо чтобы каждый заряд обсчитывался отдельно, поэтому данный флаг абсолютно неуместен.

вообще там сиротка перенёс систему перезарядки недавно, она достаточно удобная и здесь бы подошла.

NightDawnFox committed:
Уменьшение спама

Теперь при сотне кликов пустой палочкой, не будет сотни сообщений о перезарядке, лишь стандартное "buzzes quetly"

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into wand-of-slipping

NightDawnFox committed:
Улучшение кода

Меняем цифры на логические значения, добавляем уникальное сообщение при попытке выстрелить без зарядов, прячем таймер под зарядку

Gottfrei commented on code/modules/projectiles/guns/magic/wand.dm:
```diff
@@ -162,3 +162,42 @@
+
+/obj/item/gun/magic/wand/slipping/afterattack(atom/target, mob/living/user)
+	. = ..()
+	if(!charges)
+		if(!charging)
```
```suggestion
	if(!charges && !charging)
```

NightDawnFox committed:
Update code/modules/projectiles/guns/magic/wand.dm

Co-authored-by: Zwei <35403274+Gottfrei@users.noreply.github.com>

NightDawnFox committed:
багфикс
